Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/CurvePointer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Physics;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Extends line pointer to support curves. Useful for teleportation or other situations where multiple raysteps need to be tested along a spline
+    /// </summary>
+    [AddComponentMenu("Scripts/MRTK/SDK/CurvePointer")]
```

Comment:
I'm starting to notice these "AddComponentMenu" lines pop up -- do we have coding guidelines / guidance for when to add this? Should this be for every UX control? @thalbern  curious to hear your thoughts.

Reply:
The AddComponentMenu were added because of bug #6567 

They should be on anything that is a MonoBehaviour/C# script and thus can be added into the scene. 

They are in our guidelines
https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#adding-new-monobehaviour-scripts